fhir-server/src/main/java/au/csiro/pathling/fhirpath/element/DecimalPath.java

```diff
@@ -146,7 +151,7 @@ public static Function<Numeric, NonLiteralPath> buildMathOperation(@Nonnull fina
           valueColumn = valueColumn.cast(getDecimalType());
           return ElementPath
               .build(expression, dataset, idColumn, eidColumn, valueColumn, true, Optional.empty(),
-                  thisColumn, fhirType);
+                  thisColumn, source.getFhirType());
         case MODULUS:
           valueColumn = valueColumn.cast(DataTypes.LongType);
           return ElementPath
```

Comment:
Perhaps can be delegated to a counterpart of Comparator for arithmetic operations.

Reply:
Ignored (not obvious how that could be done).